### ISSUE TITLE
[Question] Correct way to control via systemd-unit-instances?

### ISSUE THREAD

**Issue by Joachim-42he** [resque/resque#1727]
I would like to move control of resque queues and number of workers per queue to systemd. 

For this I would like to create instances of simple systemd units which start one resque-worker each.
For example, If I need 3 workers for the queue "examplequeue", I would like to start 3 instances of a service unit `resque_worker_examplequeue@.service` via `systemctl start resque_worker_examplequeue@{0..2}.service`.
Inside the unit file, `ExecStart`would start `rake resque:work`.

Is there a "correct" way to do this? Since I did not have any luck with this until now: 
If I start a number of instances like this, they fail immediately. If I include `KillMode=process`the instances start and behave normally, but no longer react to `systemctl stop`or `systemctl restart`, only `systemd kill`will kill them (immediately).

To summarize: I would like to controll the forking, number of workers per queue and queues being worked on via systemd, **not** start resque via systemd and let it handle everything else itself.

**Comment by Joachim-42he** [comment]
I think I have found a solution myself. It seems I had a problem with the `backgroud` option for resque that led to the described behaviour. If that tests out, I will post my systemd-service file here for documentation purposess and close the issue.

**Comment by Joachim-42he** [comment]
Unfortunately, this dows not solve my problem here:

- If I set `background` to "1", a `systemctl stop resque_worker_examplequeue@{0..2}.service`will not stop the services at all.

- If I do not set it, Systemd indeed waits for the Resque worker to finish its current job (when `stop`or restart`are called, but this means that the systemctl command does not return until after the job is finished. Since unfortunately I have some very ling running resque-Jobs that would mean that the SSH connection to the server has to stay open (and stable) for a very long time.

**Comment by schainks** [comment]
Have you tried systemd's `--no-blocking` flag for this use case?

**Comment by Joachim-42he** [comment]
Indeed I hadn't, but this issue is moot now (for me) as we will be switching to sidekick instead. Sorry for not closing it sooner.
